pkg/vgpu/config.go

```diff
@@ -45,6 +56,152 @@ func NewNvlibVGPUConfigManager() Manager {
 	return &nvlibVGPUConfigManager{nvlib.New()}
 }
 
+func (m *nvlibVGPUConfigManager) GetVGPUConfigforVFIO(gpu int) (types.VGPUConfig, error) {
+	nvdevice, err := m.nvlib.Nvpci.GetGPUByIndex(gpu)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get GPU by index %d: %v", gpu, err)
+	}
+	vgpuConfig := types.VGPUConfig{}
+	vfnum := 0
+	if nvdevice.SriovInfo.PhysicalFunction == nil {
+		return vgpuConfig, nil
+	}
+	totalVF := int(nvdevice.SriovInfo.PhysicalFunction.TotalVFs)
+	for vfnum < totalVF {
+		vfAddr := HostPCIDevicesRoot + "/" + nvdevice.Address + "/virtfn" + strconv.Itoa(vfnum) + "/nvidia"
+		if _, err := os.Stat(vfAddr); err != nil {
+			vfnum++
+			continue
+		}
```

Comment:
Why are we incrementing `vfnum` here when the directory does not exist? As I said above, we can just use the number of VFs already calculated when constructing the NVIDIA PCI device using go-nvlib.